packages/core/agent/src/forever/forever-daemon.ts

```diff
@@ -59,35 +60,44 @@ export class ForeverDaemon implements Daemon {
       mkdirSync(logDir, { recursive: true });
       log('starting...', { profile, logDir });
 
+      const daemonLogFile = path.join(logDir, 'daemon.log');
+      const outFile = path.join(logDir, 'out.log');
+      const errFile = path.join(logDir, 'err.log');
+
+      // Clear err file.
+      fs.unlinkSync(errFile);
+
       // Run the `dx agent run` CLI command.
       // TODO(burdon): Call local run services binary directly (not via CLI)?
       forever.startDaemon(process.argv[1], {
         args: ['agent', 'start', '--foreground', `--profile=${profile}`, '--socket'],
         uid: profile,
-        logFile: path.join(logDir, 'daemon.log'), // Forever daemon process.
-        outFile: path.join(logDir, 'out.log'), // Child stdout.
-        errFile: path.join(logDir, 'err.log'), // Child stderr.
+        logFile: daemonLogFile, // Forever daemon process.
+        outFile, // Child stdout.
+        errFile, // Child stderr.
       });
-    }
 
-    // TODO(burdon): Display to user the error file.
-    const { path: socketFile } = parseAddress(getUnixSocket(profile));
-    await waitFor({
-      condition: async () => fs.existsSync(socketFile),
-      timeout: DAEMON_START_TIMEOUT,
-    });
+      await sleep(50);
+
+      const stream = await printFile(errFile);
+      const { path: socketFile } = parseAddress(getUnixSocket(profile));
+      await waitFor({
+        condition: async () => fs.existsSync(socketFile),
```

Comment:
I dont think that's enough. I'd wanna wait for a successful connection to client-services in the socket and then for the SystemService there to reply with an OK status